src/ktest/drivers/completing_internals_driver.clj

```diff
@@ -42,12 +42,7 @@
 
   (pipe-input
     [_ topic message]
-    (let [initial-result (pipe-input driver topic message)
-          {:keys [real repartitions]} (split-output initial-result)]
-      (process-messages-to-completion 1 opts driver
-                                      (flatten-output repartitions)
-                                      real)))
-
+    (process-messages-to-completion 0 opts driver [(assoc message :topic topic)] {}))
```

Comment:
Could you explain this change to me?

Reply:
Before, we did the first stage of the processing here, and passed the results into `process-messages-to-completion`, which would recursively process each layer of messages.

Now, it just passes in the initial message to `process-messages-to-completion`, and allows it to manage the first stage of the processing the same way it manages every other stage.

Reply:
ah gotcha, gather we have no test to explain why the first process was done here.

Reply:
I believe the behaviour is exactly the same, this was just duplicated code really. All the tests still pass which suggests I'm right when I think I haven't changed how it works